### PR TITLE
Remove Istio ambient compatibility blurb

### PR DESCRIPTION
**Pull request by bleggett**
Currently the [Istio integration page](https://docs.cilium.io/en/latest/network/servicemesh/istio/) states the following:

> The new experimental [ambient mesh](https://istio.io/latest/blog/2022/introducing-ambient-mesh/) data plane is not supported, as it interferes with the Cilium data plane.

[As of Istio release 1.21, this is no longer accurate](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/#support-for-all-cnis-in-ambient-mode), so removed that  blurb and lightly reworked other bits of the doc.

### CODE REVIEW

File: Documentation/network/servicemesh/istio.rst
Comment (danehans) [inline Documentation/network/servicemesh/istio.rst]: grammar nit: s/are/is/

File: Documentation/network/servicemesh/istio.rst
Comment (danehans) [inline Documentation/network/servicemesh/istio.rst:50]: I'm interested in getting input from others on the preferred UX for changing the configmap. Instead, should we lead users to [cilium CLI](https://docs.cilium.io/en/stable/configuration/), e.g. `cilium config set`?

File: Documentation/network/servicemesh/istio.rst
Comment (danehans) [inline Documentation/network/servicemesh/istio.rst]: nit: s/for details on Istio L7 policy controls./for additional details./

**Review by danehans — CHANGES_REQUESTED** [review]
@bleggett thanks for updating the doc. I have a few nits but LGTM otherwise.

File: Documentation/network/servicemesh/istio.rst
Comment (dhawton) [reply Documentation/network/servicemesh/istio.rst:50]: Probably also worth discussing managed k8s (ie GKE, DOKS, etc) where the CLI may not be a good approach/overridden by the provider.

Comment (learnitall) [reply Documentation/network/servicemesh/istio.rst:50]: From my understanding, the preferred approach is anything helm-based, ie `cilium install` or `helm upgrade --install`. This ties into the installation methods that we have in the Getting Started guide.

Comment (bleggett) [reply Documentation/network/servicemesh/istio.rst:50]: both are probably needed, but for now I just used the existing convention.

File: Documentation/network/servicemesh/istio.rst
Comment (learnitall) [inline Documentation/network/servicemesh/istio.rst]: Grammar nit:

```suggestion
* Cilium is already installed with the ``socketLB.hostNamespaceOnly`` and ``cni.exclusive=false`` Helm values.
```

File: Documentation/network/servicemesh/istio.rst
Comment (learnitall) [inline Documentation/network/servicemesh/istio.rst]: Can you wrap this line to make it smaller?

**Review by learnitall — CHANGES_REQUESTED** [review]
Thank you! A couple of small nits, otherwise LGTM.

File: Documentation/network/servicemesh/istio.rst
Comment (networkop) [inline Documentation/network/servicemesh/istio.rst:46]: has ambient mesh been tested with L4 CNPs? If traffic between a pod and a proxy does not egress through a pod's veth, no network policies (CNP or KNP) can be enforced

Comment (bleggett) [reply Documentation/network/servicemesh/istio.rst:46]: Yes, it has. As of 1.21 all Istio ambient traffic will ingress/egress through the pod normally - from the pod network namespace, using the pod's CNI-configured interfaces.

NetPol enforced in the host/node net namespace will apply as you'd expect - just as it would have with sidecars.

Comment (networkop) [reply Documentation/network/servicemesh/istio.rst:46]: is this what's described in https://istio.io/latest/blog/2024/inpod-traffic-redirection-ambient/#istio-ambient-traffic-redirection-the-new-model?

Comment (bleggett) [reply Documentation/network/servicemesh/istio.rst:46]: Yes.

Comment (networkop) [reply Documentation/network/servicemesh/istio.rst:46]: but if I read it correctly, all traffic redirected into a ztunnel proxy will bypass host netNS and, thus, any network policy applied to a pod

Comment (bleggett) [reply Documentation/network/servicemesh/istio.rst:46]: No, all traffic will be captured once it enters a pod, and before it leaves a pod.

It will always pass through the host netns, and thus always be subject to CNI NetPol.

The old method that was replaced bypasses the host netns, the new method does not.

Comment (networkop) [reply Documentation/network/servicemesh/istio.rst:46]: sorry, I haven't been clear enough. As the traffic leaves the pod, and gets redirected to ztunnel, it will get encrypted, so by the time it hits host netNS it's original src/dst ports are no longer the same, right?

Comment (bleggett) [reply Documentation/network/servicemesh/istio.rst:46]: Yes, for mTLS traffic leaving the pod, it will be tunneled and mTLS encrypted over port 15008 with an HTTP/2 tunnel.

NetPol would still apply to that traffic just as it would if the pod natively tunneled with Wireguard or any other tunnel, but the egress port would change in that case. 

The original ports would be retained inside that tunnel and the packet would (eventually) be routed accordingly, but the host netns wouldn't see that.

Once the traffic is tunneled and encrypted, the contents of the encrypted tunnel are naturally not visible to things in the host netns.

Ingress would not be affected either way.

This is something we can explain in detail here, or in the Istio docs, or both.

This would actually be covered by the existing instructions to globally disable Istio mTLS, which is why I left that in - with no peer authentication,  there would be no tunneling and therefore the original ports would be retained on both cases.

The existing advice to globally disable Istio mTLS encryption and peer auth is imo somewhat extreme advice now with in-pod ambient capture, as that does actually lessen the need to globally disable mTLS in Istio ambient overall versus sidecar - since with ambient in Istio 1.21 if you left strict mTLS auth ON, all you would lose is egress port targeting via NetPol - you'd have to do the equivalent in Istio policy instead (or just change the ports, depending), and if you weren't using egress port targeting NetPol at all (that's pretty uncommon from what we see, mostly it's just ingress control), you wouldn't care.

But again, I left in the existing "disable globally" text to remain consistent and not scope-creep this PR too much.

File: Documentation/network/servicemesh/istio.rst
Comment (danehans) [inline Documentation/network/servicemesh/istio.rst]: @bleggett can you help clarify "When using Istio in sidecar mode with Kubernetes admission webhooks"? From my understanding, this is Istio sidecar mode with [automatic sidecar injection](https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/#automatic-sidecar-injection), correct? If so, can you update accordingly?

File: Documentation/network/servicemesh/istio.rst
Comment (danehans) [reply Documentation/network/servicemesh/istio.rst:46]: +1 on summarizing https://istio.io/latest/blog/2024/inpod-traffic-redirection-ambient/#istio-ambient-traffic-redirection-the-new-model so a reader understands how the magic happens.

File: Documentation/network/servicemesh/istio.rst
Comment (danehans) [inline Documentation/network/servicemesh/istio.rst]: `with other CNI plugins on the node` Since this doc is specific to Istio integration, should this general statement be more specific by stating "with the Istio CNI plugin"

Comment (bleggett) [reply Documentation/network/servicemesh/istio.rst]: The flag itself is general, and exists for any CNI (AWS, etc) - any other CNI plugin would have this problem without this flag, and other parts of the doc for other CNI/cloud integrations have the same guidance for the same reason.

It would be ideal if we could link back directly to this general flag and its specific documentation, which would probably help remove the redundant special-casing we have for it in other spots.

File: Documentation/network/servicemesh/istio.rst
Comment (networkop) [reply Documentation/network/servicemesh/istio.rst:46]: I think providing a summary here with links to external resources (esp. around recommendations to disable mTLS) would be enough. We just need to be clear about things that may not work with Ambient

File: Documentation/network/servicemesh/istio.rst
Comment (bleggett) [reply Documentation/network/servicemesh/istio.rst]: Reworded a bit.

File: Documentation/network/servicemesh/istio.rst
Comment (bleggett) [reply Documentation/network/servicemesh/istio.rst:46]: @danehans and @networkop PTAL, redid the wording a bit. Hopefully a bit clearer.

File: Documentation/network/servicemesh/istio.rst
Comment (networkop) [inline Documentation/network/servicemesh/istio.rst]: This would affect both egress and ingress from Cilium-managed pods, so maybe `s/egress/L4`?

File: Documentation/network/servicemesh/istio.rst
Comment (networkop) [inline Documentation/network/servicemesh/istio.rst]: > Istio must be used to enforce policy for traffic between Istio-managed workloads

can we make it a bit more concrete, e.g. ` Istio can be used to enforce policy for traffic between Istio-managed workloads at L4 and above`

**Review by networkop — CHANGES_REQUESTED** [review]

File: Documentation/network/servicemesh/istio.rst
Comment (networkop) [reply Documentation/network/servicemesh/istio.rst:46]: thanks @bleggett , left a couple of minor comments but overall the wording looks good

File: Documentation/network/servicemesh/istio.rst
Comment (bleggett) [reply Documentation/network/servicemesh/istio.rst]: Reworded the section again, agree.

File: Documentation/network/servicemesh/istio.rst
Comment (bleggett) [reply Documentation/network/servicemesh/istio.rst]: Reworded this a bit - all the pods are Cilium managed, this only applies to traffic between two Istio-managed pods with encryption+tunneling enabled.

Traffic from a non-Istio pod to an Istio pod would not be tunneled + encrypted, and the Cilium netpol would apply as-is there, without changes.

**Comment by danehans** [comment]
@bleggett CI is failing due to:
```
ERROR:NO_AUTHOR_SIGN_OFF: Missing Signed-off-by: line by nominal patch author 'Daniel Hawton <daniel@hawton.org>'
```

@dhawton will need to sign off. Since all the changes are related to docs, maintainers may also want you to squash into a single commit.

**Comment by bleggett** [comment]
> @bleggett CI is failing due to:
> 
> ```
> ERROR:NO_AUTHOR_SIGN_OFF: Missing Signed-off-by: line by nominal patch author 'Daniel Hawton <daniel@hawton.org>'
> ```
> 
> @dhawton will need to sign off. Since all the changes are related to docs, maintainers may also want you to squash into a single commit.

Fair enough, forgot cilium doesn't autosquash on merge.

Manually squashed, should fix the signoff lint too, I think?

File: Documentation/network/servicemesh/istio.rst
Comment (danehans) [inline Documentation/network/servicemesh/istio.rst]: nit: s/fully subject/fully subjected/

**Review by danehans — APPROVED** [review]
Other than my one nit, LGTM. Thanks @bleggett for working through the review feedback.

**Review by networkop — APPROVED** [review]
lgtm. thanks

**Review by youngnick — APPROVED** [review]

**Comment by youngnick** [comment]
Change LGTM but I think that the commit message for 4a783d3 is long enough that it's making the BPF Checks CI entry fail. I'd recommend truncating to just `Add a separate section for the Istio + Cilium demo for ambient`.

**Comment by bleggett** [comment]
> Change LGTM but I think that the commit message for [4a783d3](https://github.com/cilium/cilium/commit/4a783d3e3454e3299bf7e2547998d2d2dcd5f1a3) is long enough that it's making the BPF Checks CI entry fail. I'd recommend truncating to just `Add a separate section for the Istio + Cilium demo for ambient`.

resquashed, ty!

**Review by learnitall — APPROVED** [review]
Thank you!

**Review by dhawton — APPROVED** [review]

**Comment by bleggett** [comment]
ty!

**Pull request merged**
